### ISSUE TITLE
Non-integer k and variable-Q transform

### ISSUE THREAD

**Issue by TF3RDL** [creaktive/pianolizer#5]
Turns out, it is possible to have non-integer k on CQ-sDFT algorithm, which means analysis frequency can exactly corresponds to musical notes instead of approximate one (as mentioned in #3) and it also makes it easier to convert into variable-Q transform like [this](https://gist.github.com/TF3RDL/1810c570bad4f75f77941c981c65723f), it just need one additional factor, namely fiddles

**Comment by creaktive** [comment]
Hey, this is amazing! Thanks for sharing!
And big thanks for doing the implementation :)
May I link your gist from the references section of my project?

**Comment by TF3RDL** [comment]
> May I link your gist from the references section of my project?

Yes, for somebody besides you and me who make foobar2000 components (by using my JS project as a reference and re-implement the same algorithm in C++ for use in foobar2000 visualization components), especially the fact a lot of components like Musical Spectrum are no longer developed (thus no 64-bit version for use in foobar2000 x64) and are only usable within 32-bit version of foobar2000

**Comment by TF3RDL** [comment]
to add to this, I've made [this CodePen project](https://codepen.io/TF3RDL/pen/MWLzPoO) demonstrating the sliding DFT with noninteger K allowing variable-Q transform with smoother low frequency peaks and also a faster IIR cousin, the sliding windowed infinite Fourier transform (SWIFT) and even analog-style filter bank (which some professional RTAs including Spectralissime uses it over FFT) and a [YouTube video demoing the good-old algorithm](https://youtube.com/watch?v=K_05fdSt8Y0) not based on FFT
